### ISSUE TITLE
[PATCH] twisted.protocols.portforward has a couple of problems.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7078]
|<img alt="Omnifarious's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Omnifarious reported|
|-|-|
|Trac ID|trac#527|
|Type|defect|
|Created|2004-02-26 09:41:43Z|

Attachments:

* [pf.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/22a/22ab0e4326f1e9096ffc3217cd20a17829a5c486/24f8b65cbfc28d4fc01f9e6c10fc990b584f1145.patch) (1867 bytes) - added by Omnifarious on 2004-02-26 09:41:43Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__527 527
type__defect defect
reporter__Omnifarious Omnifarious
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1077788503000000 1077788503000000
changetime__1472251873724337 1472251873724337
version__ 
owner__teratorn teratorn
cc__teratorn cc__Omnifarious
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Omnifarious's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Omnifarious commented|
|-|-|

```
#!html
<pre>
twisted.protocols.portforward has a few small problems that I have a patch for.

First, if it loses the connection from the thing that connected to it before it
makes the outgoing connection, and then fails to make the outgoing connection,
it throws a nasty exception and dies.

Second, it used to buffer data from the thing connecting to it until it made an
outgoing connection, then it would dump all the data into the outgoing
connection.  This is the wrong behavior.  It shouldn't buffer anything.  I was
able to get it to buffer 45M of data before the outgoing connection failed and
shut down.

The included file is a diff to portforward.py that should fix the problem.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn commented|
|-|-|

```
#!html
<pre>
I'll take a look at this when I get a chance.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
Slightly modified patch, plus unit tests, applied.  Resolved in r11826.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc commented|
|-|-|

In changeset 011e7d7be7cd00827a8664dfd3fe6271558746d6
```
#!CommitTicketReference repository="" revision="011e7d7be7cd00827a8664dfd3fe6271558746d6"
Merge pull request #527 from shyba/8788-adbapi-typo

Author: shyba
Reviewer: rodrigc
Fixes: #8788

Fix typo
```
